src/commit.ts

```diff
@@ -326,9 +326,24 @@ function hasExtendedContext(line: string) {
   return false;
 }
 
-function parseCommits(message: string): parser.ConventionalChangelogCommit[] {
+function parseOrUndefined(commit: Commit, message: string): parser.Message {
+  try {
+    return parser.parser(message);
+  } catch (_err) {
```

Comment:
Does this throw? If it does, then the existing code would have a bad time with non-conforming commits.

Reply:
Yeah, parser throws an exception. I thought pretty long, whether change should be done in the parser itself, but I don't think it really belongs there (or it would have to be hidden by an optional flag).

Despite the hack-ish implementation, it seems to work pretty well here.